lms/static/js/learner_dashboard/views/entitlement_unenrollment_view.js

```diff
@@ -0,0 +1,149 @@
+                closeButtonSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-close-btn',
+                trackInfoSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-track-info',
+                refundInfoSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-refund-info',
+                errorInfoSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-error-info',
+                submitButtonSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-submit',
+                triggerSelector: '.js-entitlement-action-unenroll',
+                mainPageSelector: '#dashboard-main',
+
+                initialize: function(options) {
+                    var view = this;
+                    this.dashboardPath = options.dashboardPath;
+                    this.signInPath = options.signInPath;
+
+                    this.$submitButton = $(this.submitButtonSelector);
+                    this.$trackInfo = $(this.trackInfoSelector);
+                    this.$refundInfo = $(this.refundInfoSelector);
+                    this.$errorInfo = $(this.errorInfoSelector);
+
+                    this.$submitButton.on('click', this.handleSubmit.bind(this));
+
+                    $(this.triggerSelector).each(function() {
+                        var $trigger = $(this);
+
+                        $trigger.on('click', view.handleTrigger.bind(view));
+
+                        if (window.accessible_modal) {
+                            window.accessible_modal(
+                                '#' + $trigger.attr('id'),
+                                view.closeButtonSelector,
+                                '#' + view.$el.attr('id'),
+                                view.mainPageSelector
+                            );
+                        }
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
+                        courseNumber = $trigger.data('courseNumber'),
+                        isRefundable = $trigger.data('entitlementIsRefundable'),
+                        apiEndpoint = $trigger.data('entitlementApiEndpoint');
+
+                    // Convert to boolean.
+                    isRefundable = isRefundable && isRefundable.toLowerCase() === 'true';
+
+                    this.resetModal();
+                    this.setTrackInfo(courseName, courseNumber);
+                    this.setRefundInfo(isRefundable);
+                    this.setSubmitData(apiEndpoint);
+
+                    if (window.edx && window.edx.dashboard && window.edx.dashboard.dropdown) {
+                        window.edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+                        this.$el.css('position', 'fixed');
```

Comment:
We shouldn't be using JS to add inline styles. It's preferable to add CSS classes that contain the styles.

Reply:
In this case, I think we need to add the styles directly because I believe `leanModal`, or something, is adding styles directly to the element. If we just apply a class here, I believe the direct styles added by the other JS code would apply.

Reply:
Thanks for the clarification.